### ISSUE TITLE
[reggen] Assumptions with countermeasures blocking rv_dm V1 progress

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#10872]
RV_DM implements a custom hjson parsed by reggen for DV purposes. See b87e545d2b7943a9766d3defebbdee3cbc6d6104 for details. 

Unfortunately the checks in the reggen codebase surrounding countermeasures results in errors being thrown. 

Command:
```bash
./util/regtool.py -s -t . hw/ip/rv_dm/dv/env/../../data/rv_dm_debug_mem.hjson
```

If the list of countermeasures are removed altogether, the following error is thrown:
```
RuntimeError: One or more unknown countermeasures referenced in RTL.
```

If the following countermeasure is added to it:
```
  countermeasures: [
    { name: "BUS.INTEGRITY",
      desc: "End-to-end bus integrity scheme."
    }
  ]
```

It now throws this error:
```
RuntimeError: One or more unknown countermeasures referenced in RTL.
```

This seems to be a new update. 

Please suggest a fix.

**Comment by sriyerg** [comment]
IMO the whole countermeasure creation / check should happen only when dumping RTL, i.e. when `format == 'rtl'`.

**Comment by sriyerg** [comment]
I implemented a fix for this here:
https://github.com/lowRISC/opentitan/pull/10858/commits/64d6c581bffc06de0cadf78d0977d98fbf7eb86b#r807576111

PTAL, thanks!

**Comment by msfschaffner** [comment]
Ah I see. 

Yes, we added checks that go both ways (so if there are annotations in RTL that are not accounted for in the hjson, it will throw an error).

I think we made the check unconditional, because at that time this did not cause issues, and gives more coverage (so to speak). But I am fine with predicating this on RTL generation, since this check pertains to the RTL files only.

See https://github.com/lowRISC/opentitan/pull/10080#discussion_r784754848
